on_node_configured.sh

```diff
@@ -0,0 +1,219 @@
+#!/usr/bin/env bash
+
+# https://gist.github.com/mohanpedala/1e2ff5661761d3abd0385e8223e16425#file-bash_strict_mode-md
+set -euo pipefail
+
+#####
+# Script arguments
+#####
+SLURM_JWT_KEY="$1"
+# GitHub release tag for UQLE CLI tool
+# CLI_TAG="$2"
+# GitHub OAuth token - should have read access to UQLE CLI releases, and the UQLE stack repository
+# MACHINE_USER_TOKEN="$3"
+# UQLE_API_HOST="$4"
+
+# global variables
+JWT_KEY_DIR=/var/spool/slurm.state
+JWT_KEY_FILE="$JWT_KEY_DIR"/jwt_hs256.key
+SLURMDBD_USER="slurm"
+SLURM_PASSWORD_FILE=/root/slurmdb.password
+
+. /etc/parallelcluster/cfnconfig
+
+echo "Node type: $cfn_node_type"
+
+function modify_slurm_conf() {
+    # add JWT auth and accounting config to slurm.conf
+    # /opt/slurm is shared via nfs, so this only needs to be configured on head node
+
+    cat >>/opt/slurm/etc/slurm.conf <<EOF
+# Enable jwt auth for Slurmrestd
+AuthAltTypes=auth/jwt
+# ACCOUNTING
+JobAcctGatherType=jobacct_gather/linux
+AccountingStorageType=accounting_storage/slurmdbd
+AccountingStorageHost=localhost
+AccountingStoragePort=6819
+EOF
+}
+
+function configure_users_common() {
+    sysctl user.max_user_namespaces=15000
+    usermod --add-subuids 165536-231071 --add-subgids 165536-231071 slurm
+}
+
+function configure_users_head_node() {
+    configure_users_common
+
+    cat <<'EOF' | tee -a /home/centos/.bashrc /home/slurm/.bashrc
+# Set variables to avoid podman conflicts between nodes due to nfs-sharing of /home
+# See basedir-spec at https://specifications.freedesktop.org/
+
+base_xdg_dir=$(mktemp -qd /tmp/"$(id -u)"-xdg-XXXXXXXXXX)
+
+export XDG_RUNTIME_DIR="$base_xdg_dir"/.runtime
+export XDG_DATA_HOME="$base_xdg_dir"/.data
+export XDG_STATE_HOME="$base_xdg_dir"/.state
+export XDG_CACHE_HOME="$base_xdg_dir"/.cache
+export XDG_CONFIG_HOME="$base_xdg_dir"/.config
+
+unset base_xdg_dir
+
+for directory in {"$XDG_RUNTIME_DIR","$XDG_DATA_HOME","$XDG_STATE_HOME","$XDG_CACHE_HOME","$XDG_CONFIG_HOME"}; do
+    mkdir -p "$directory"
+done
+
+alias podman="podman --runroot=\$XDG_RUNTIME_DIR --root=\$XDG_DATA_HOME"
+EOF
+}
+
+function write_jwt_key_file() {
+    # set the jwt key
+    if [ "$SLURM_JWT_KEY" ]; then
+        echo "- JWT secret variable found, writing..."
+
+        mkdir -p "$JWT_KEY_DIR"
+
+        echo -n "$SLURM_JWT_KEY" >"$JWT_KEY_FILE"
+    else
+        echo "Error: JWT key not present in environment - aborting cluster deployment" >&2
+        return 1
+    fi
+
+    chown slurm:slurm "$JWT_KEY_FILE"
+    chmod 0600 "$JWT_KEY_FILE"
+}
+
+function create_slurmrest_conf() {
+    # create the slurmrestd.conf file
+    # this file can be owned by root, because the slurmrestd service is run by root
+    cat >/opt/slurm/etc/slurmrestd.conf <<EOF
+include /opt/slurm/etc/slurm.conf
+AuthType=auth/jwt
+EOF
+
+}
+
+function create_slurmdb_conf() {
+    local slurmdbd_password
+    slurmdbd_password=$(cat "$SLURM_PASSWORD_FILE")
```

Comment:
*non-blocking, query:* These two lines could just be on one line, right? Not that it makes any difference, just checking.

Reply:
A 1-liner here would introduce a [subtle issue](https://www.shellcheck.net/wiki/SC2155), in that if `cat "$SLURM_PASSWORD_FILE"` fails, then this line would still return true, because the exit status of the variable creation is what is returned. This even affects `set -e`.

Reply:
You learn something every day